components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/SolaceAdminApis.java

```diff
@@ -141,6 +141,84 @@ public CloseableHttpResponse registerAPI(String organization, String title, Stri
         return null;
     }
 
+    /**
+     * Update API in Solace using AsyncAPI Definition
+     *
+     * @param organization  name of the Organization
+     * @param title         name of the Solace API
+     * @param apiDefinition Async definition of the Solace API
+     * @return CloseableHttpResponse of the PUT call
+     */
+    public CloseableHttpResponse updateAPIDefinition(String organization, String title, String apiDefinition) {
+        URL serviceEndpointURL = new URL(baseUrl);
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPatch httpPatch = new HttpPatch(baseUrl + "/" + organization + "/apis/" + title);
+        httpPatch.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getBase64EncodedCredentials());
+        httpPatch.setHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
+        JsonNode jsonNodeTree;
+        String jsonAsYaml = null;
+        // convert json to yaml
+        try {
+            jsonNodeTree = new ObjectMapper().readTree(apiDefinition);
+            jsonAsYaml = new YAMLMapper().writeValueAsString(jsonNodeTree);
+        } catch (JsonProcessingException e) {
+            log.error(e.getMessage());
+        }
+        //add definition to request body
+        if (jsonAsYaml != null) {
+            StringEntity params = null;
+            try {
+                params = new StringEntity(jsonAsYaml);
+            } catch (UnsupportedEncodingException e) {
+                log.error(e.getMessage());
+            }
+            httpPatch.setEntity(params);
+            try {
+                return APIUtil.executeHTTPRequest(httpPatch, httpClient);
+            } catch (IOException | APIManagementException e) {
+                log.error(e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Update API Product in Solace using AsyncAPI Definition
+     *
+     * @param organization           name of the Organization
+     * @param environment            name of the Environment
+     * @param apiProductName         name of the API product
+     * @param apiNameForRegistration name of the Solace API product
+     * @param aai20Document          Async definition of the Solace API
+     * @return CloseableHttpResponse of the POST call
```

Comment:
Shall we call the Call as a PATCH call?

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11001/commits/9c9fe10fbb271f44e424d8a7efc0a76be0fc7b09

Reply:
Yes it is updated.